.github/workflows/static-analysis.yml

```diff
@@ -47,3 +52,19 @@ jobs:
         uses: "github/codeql-action/upload-sarif@v3"
         with:
           sarif_file: psalm.sarif
+          ref: ${{ github.event_name == 'workflow_dispatch' && inputs.ref || github.ref }}
+          sha: ${{ github.event_name == 'workflow_dispatch' && env.CHECKED_OUT_SHA || github.sha }}
+
+      - name: "Upload SARIF report"
+        if: ${{ github.event_name != 'workflow_dispatch' }}
```

Comment:
I needed to split this, as the upload-sarif action contains logic to fix the `refs/pulls/<number>/merge` ref, but doesn't do so when it was provided explicitly. Thus the only way to retain current logic for pull requests is to invoke the action without `ref` and `sha` arguments, and then explicitly pass them when the workflow was dispatched manually.